### PR TITLE
Support code blocks

### PR DESCRIPTION
**Pull request by connyduck**
before

after

### CODE REVIEW

File: app/src/main/java/com/keylesspalace/tusky/util/StatusParsingHelper.kt [hunk 1]
```diff
@@ -17,21 +17,78 @@
...
 /**
  * parse a String containing html from the Mastodon api to Spanned
  */
 @JvmOverloads
-fun String.parseAsMastodonHtml(tagHandler: TagHandler? = null): Spanned {
+fun String.parseAsMastodonHtml(tagHandler: TagHandler? = tuskyTagHandler): Spanned {
     return this.replace("<br> ", "<br>&nbsp;")
         .replace("<br /> ", "<br />&nbsp;")
         .replace("<br/> ", "<br/>&nbsp;")
+        .replace("\n", "<br/>")
```
Comment (Tak) [inline app/src/main/java/com/keylesspalace/tusky/util/StatusParsingHelper.kt:37]: Do we need to inject the nonbreaking space here? Or we only get `\n` when it's a code block?

**Review by Tak — APPROVED** [review]

File: app/src/main/java/com/keylesspalace/tusky/util/StatusParsingHelper.kt (hunk 1 above)
Comment (connyduck) [reply app/src/main/java/com/keylesspalace/tusky/util/StatusParsingHelper.kt:37]: When a browser renders HTML, it ignores newlines as a default. The Android HTML parsing code does the same.
In a browser, you can override that behavior with the [whitespace css property](https://developer.mozilla.org/en-US/docs/Web/CSS/white-space). Mastodon sets it for the whole status to `pre-wrap`. That difference did not matter, because Mastodon also replaces all newlines with `<br/>` so the Android parser would handle them correctly. But inside `<code>` blocks, newlines are not replaced by Mastodon, so that's what this replacement is for.

Tldr:
> we only get \n when it's a code block?

yes
